src/api/courses.js

```diff
@@ -69,6 +69,124 @@ router.get(
   })
 )
 
+// Get course queue data
+router.get(
+  '/:courseId/data',
+  [requireCourseStaff, requireCourse, failIfErrors],
+  safeAsync(async (req, res, _next) => {
+    const { id: courseId } = res.locals.course
+    const questions = await Question.findAll({
+      include: [
+        {
+          model: Queue,
+          include: [
+            {
+              model: Course,
+              attributes: [['name', 'CourseName']],
+              required: true,
+              where: { courseId: Sequelize.col('queue.courseId') },
+            },
+          ],
+          attributes: [
+            ['id', 'queueId'],
+            'courseId',
+            ['name', 'QueueName'],
+            ['location', 'QueueLocation'],
+            [
+              Sequelize.fn(
+                'datetime',
+                Sequelize.col('queue.createdAt'),
+                'localtime'
+              ),
+              'Queue_CreatedAt',
+            ],
+          ],
+          required: true,
+          where: { courseId, id: Sequelize.col('question.queueId') },
+        },
+        {
+          model: User,
+          as: 'askedBy',
+          attributes: {
+            include: [
+              ['netid', 'AskedBy_netid'],
+              ['universityName', 'AskedBy_RealName'],
```

Comment:
Calling something a "real" name has certain unwanted connotations. For instance, I have trans friends whose university name is not their real name; there are also students who use a different name day-to-day from what's in Enterprise. I think there are a couple potential changes here:

* calling this `UniversityName`
* the above, but also exporting the `PreferredName` field
* doing what the UI/API already do: taking the preferred name if it's specified and falling back to the name provided by Shib if they aren't
* omitting the name entirely

I think my personal preference would be the third option: it's most consistent with how names are treated everywhere else in the queue. There is the potential for abuse (I can set my preferred name to an arbitrary string), but the nedid can always be cross-referenced with a roster or the university directory if there are doubts about a user's identity.

Reply:
I'll play around with the options you listed here - maybe one of them can solve the weird issue mentioned above.